src/test/java/com/thealgorithms/datastructures/graphs/WelshPowellAlgorithmTest.java

```diff
@@ -0,0 +1,47 @@
+package com.thealgorithms.datastructures.graphs;
+
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+import org.junit.jupiter.api.Test;
+
+class WelshPowellAlgorithmTest {
+
+    @Test
+    void testSimpleGraph() {
+        WelshPowell.WPGraph graph = WelshPowell.createGraph(4);
+        WelshPowell.addEdge(graph, 0, 1);
+        WelshPowell.addEdge(graph, 1, 2);
+        WelshPowell.addEdge(graph, 2, 3);
+        int[] colors = WelshPowell.welshPowellColoring(graph);
+        assertTrue(isValidColoring(graph, colors));
```

Comment:
Imagine that the `WelshPowell.welshPowellColoring` would simply return an array of _unique integers_ (like `{0, 1, 2, ..., n-1}`), then each of these tests would pass.

Reply:
currently working on  making the  test class according to this.